api/kogito-legacy-api/src/main/java/org/kie/kogito/legacy/rules/impl/KieSessionImpl.java

```diff
@@ -0,0 +1,398 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.legacy.rules.impl;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.kie.api.KieBase;
+import org.kie.api.command.Command;
+import org.kie.api.event.process.ProcessEventListener;
+import org.kie.api.event.rule.AgendaEventListener;
+import org.kie.api.event.rule.RuleRuntimeEventListener;
+import org.kie.api.logger.KieRuntimeLogger;
+import org.kie.api.runtime.Calendars;
+import org.kie.api.runtime.Channel;
+import org.kie.api.runtime.Environment;
+import org.kie.api.runtime.Globals;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.KieSessionConfiguration;
+import org.kie.api.runtime.ObjectFilter;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.kie.api.runtime.rule.Agenda;
+import org.kie.api.runtime.rule.AgendaFilter;
+import org.kie.api.runtime.rule.EntryPoint;
+import org.kie.api.runtime.rule.FactHandle;
+import org.kie.api.runtime.rule.LiveQuery;
+import org.kie.api.runtime.rule.QueryResults;
+import org.kie.api.runtime.rule.ViewChangedEventListener;
+import org.kie.api.time.SessionClock;
+
+public class KieSessionImpl implements org.kie.kogito.legacy.rules.KieSession {
+
+    private final KieSession delegate;
+
+    public KieSessionImpl(org.kie.api.runtime.KieSession delegate) {
+        this.delegate = delegate;
+    }
+
+    public KieSession getDelegate() {
+        return delegate;
+    }
+
+    @Override
+    @Deprecated
+    public int getId() {
+        return delegate.getId();
+    }
+
+    @Override
+    public long getIdentifier() {
+        return delegate.getIdentifier();
+    }
+
+    @Override
+    public void dispose() {
+        delegate.dispose();
+    }
+
+    @Override
+    public void destroy() {
+        delegate.destroy();
+    }
+
+    @Override
+    public void submit(AtomicAction atomicAction) {
+        delegate.submit(atomicAction);
+    }
+
+    @Override
+    public <T> T getKieRuntime(Class<T> aClass) {
+        return delegate.getKieRuntime(aClass);
+    }
+
+    @Override
+    public int fireAllRules() {
+        return delegate.fireAllRules();
+    }
+
+    @Override
+    public int fireAllRules(int i) {
+        return delegate.fireAllRules(i);
+    }
+
+    @Override
+    public int fireAllRules(AgendaFilter agendaFilter) {
+        return delegate.fireAllRules(agendaFilter);
+    }
+
+    @Override
+    public int fireAllRules(AgendaFilter agendaFilter, int i) {
+        return delegate.fireAllRules(agendaFilter, i);
+    }
+
+    @Override
+    public void fireUntilHalt() {
+        delegate.fireUntilHalt();
+    }
+
+    @Override
+    public void fireUntilHalt(AgendaFilter agendaFilter) {
+        delegate.fireUntilHalt(agendaFilter);
+    }
+
+    @Override
+    public <T> T execute(Command<T> command) {
```

Comment:
I'm not sure we want to preserve `execute` method but at least let's improve it using `ExecutableCommand` interface :)
```suggestion
    public <T> T execute(ExecutableCommand<T> command) {
```

Reply:
+1 to throw UOE